File: hh-examples/03-3-harnessing-jpa/src/main/java/com/oreilly/hh/CreateTest.java
package com.oreilly.hh;

import javax.persistence.*;

import com.oreilly.hh.data.*;

import java.time.LocalTime;
import java.util.*;

/**
 * Create sample data, letting Hibernate persist it for us.
 */
public class CreateTest {

    public static void main(String args[]) throws Exception {

        EntityManagerFactory emf = Persistence.createEntityManagerFactory("default");
        EntityManager        em  = emf.createEntityManager();
        EntityTransaction    tx  = em.getTransaction();

        try {
            tx.begin();

            Track track = new Track("Russian Trance",
                                    "vol2/album610/track02.mp3",
                                    LocalTime.of(00,03,30),
                                    (short)0);
            em.persist(track);

            track = new Track("Video Killed the Radio Star",
                              "vol2/album611/track12.mp3",
                              LocalTime.of(00,03,49),
                              (short)0);
            em.persist(track);

            track = new Track("Gravity's Angel",
                              "vol2/album175/track03.mp3",
                              LocalTime.of(00,06,06),
                              (short)0);
            em.persist(track);

            tx.commit();
        } catch (RuntimeException e) {
            tx.rollback();
            throw e;
        } finally {
            em.close();
        }

        emf.close();
    }
}

File: hh-examples/03-4-harnessing-jpa-kotlin/src/main/kotlin/com/oreilly/hh/CreateTest.kt
package com.oreilly.hh

import java.time.LocalTime

import javax.persistence.Persistence

import com.oreilly.hh.data.*

fun main() {
    Persistence.createEntityManagerFactory("default").use { emf ->

        emf.createEntityManager().use { em ->
            em.transaction.use {
                var track = Track("Russian Trance", "vol2/album610/track02.mp3",
                        LocalTime.of(0, 3, 30))
                em.persist(track)

                track = Track("Video Killed the Radio Star", "vol2/album611/track12.mp3",
                        LocalTime.of(0, 3, 49))
                em.persist(track)

                track = Track("Gravity's Angel", "vol2/album175/track03.mp3",
                        LocalTime.of(0, 6, 6))
                em.persist(track)
            }
        }
    }
}
